scripts/run-tests.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+// As node 20 test runner does not support glob patterns in input
+// and considering that we could have multiple OS we manually
+// resolve the test files and pass them to the test runner
+
+const { spawnSync } = require('child_process');
+const { globSync } = require('glob');
+
+const testFiles = globSync('test/**/*test.ts', { absolute: true });
```

Comment:
This lgtm, the only proposal I'd like to add is to receive the pattern from the argv instead, so we have the chance to reuse this implementation across other use cases.

Maybe even the flags, but we can start with the glob pattern first

Reply:
sure 😁 I'll change it asap 🚀

Reply:
Pushed!

On another matter: I was checking the output of the coverage and it seems different.
I've been checking the results and, as far as I can tell, they seem correct even though they are different.

Could you please take a look too?

Reply:
This is an example of the new output of `coverage/lcov-report/src/histogram.ts.html`
<img width="638" alt="image" src="https://github.com/user-attachments/assets/3e9052eb-6c93-41a9-9ee3-6198d7f35589" />

Reply:
Interesting, the actual coverage shown by your PR is more accurate than the one we currently have.

The uncovered lines are correct, can we address that before landing this?